### PR TITLE
Fix dnsenum components

### PR DESCRIPTION
**Pull request by junedtk632**
The missing components were added with several key fixes and improvement to DNSenumTool, addressing errors and enhancing functionality related to command availability, process management and component initialization using useEffect. 
1. **checkAllCommandsAvailability** component added, which verifies if necessary commands dnsenum, kill are installed on the system or not. 
2. Displays an **InstallationModal** component if commands are missing, guiding use to install them manually. 
3. The **useEffect** component triggers command availability check when the component is mounted. It ensures that command validation happens automatically upon loading the tool, without requiring user interaction. 
4. Process termination issues are also fixed. Provided feedback for different termination scenarios such as successful completion, manual termination, or errors. 

image

### CODE REVIEW

**Comment by michaeljpigott** [comment]
@junedtk632 

You need to fix the conflict with DNSenumTool.tsx. I think you based this branch on an outdated version of because the tool has already been baselined. 

Kind regards

Michael

**Comment by michaeljpigott** [comment]
@junedtk632 you need to run `yarn run style` and `npx prettier --write <file path>` to fix the failed CI check.

**Comment by michaeljpigott** [comment]
@junedtk632 is this ready for review again?

**Comment by junedtk632** [comment]
Yes, it is ready to review again. I have checked there are no errors in the
build on github.

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 1]
```diff
@@ -109,6 +111,24 @@ const DnsenumTool = () => {
      * @param {FormValuesType} values - The form values, containing the domain, threads, pages, scrap, reverse lookup, and timeout.
      */
 
+    const checkAllCommandsAvailability = async () => {
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:114]: Interesting way of implementing this. I am getting the following: 

image

I recommend implementing as per our other tools.

Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:114]:
```suggestion
    // Check the availability of commands in the dependencies array
    useEffect(() => {
        checkAllCommandsAvailability(dependencies)
            .then((isAvailable) => {
                setIsCommandAvailable(isAvailable);
                setOpened(!isAvailable);
                setLoadingModal(false);
            })
            .catch((error) => {
                console.error("An error occurred:", error);
                setLoadingModal(false);
            });
    }, []);
```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 2]
```diff
@@ -109,6 +111,24 @@ const DnsenumTool = () => {
...
+            }
+        }
+        setMissingCommands(unavailableCommands);
+        setModalOpened(unavailableCommands.length > 0);
+    };
+
+    useEffect(() => {
+        checkAllCommandsAvailability();
+    }, []);
+
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:131]: I recommend replacing these lines with our conventional use of useEffect()
```suggestion
      

```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Button, LoadingOverlay, Stack, TextInput, Switch, Checkbox } from "@mantine/core";
+import { Button, LoadingOverlay, Stack, TextInput, Switch, Modal } from "@mantine/core";
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:1]: You also need to import checkAllCommandsAvailability and Installation Modal. 
```suggestion
import { Button, LoadingOverlay, Stack, TextInput, Switch, Modal } from "@mantine/core";
import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";'
import InstallationModal from "../InstallationModal/InstallationModal";

```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 4]
```diff
@@ -223,6 +243,16 @@ const DnsenumTool = () => {
                     <Button type={"submit"}>Start Enumeration</Button>
                     {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
                     <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
+
+                    <Modal opened={modalOpened} onClose={() => setModalOpened(false)} title="Missing Commands">
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:247]: Interesting implementation but I recommend that you add the conventional component installationModal. Here is an example of what it looks like in WhatWeb: 

image
```suggestion
                 
```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 5]
```diff
@@ -223,6 +243,16 @@ const DnsenumTool = () => {
...
+                        <p>The Following Commands are missing and need to be installed:</p>
+                        <ul>
+                            {missingCommands.map((cmd) => (
+                                <li key={cmd}></li>
+                            ))}
+                        </ul>
+                        <p>Please install these commands and try again.</p>
+                    </Modal>
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:255]: Remove once you add the convention installationModal component. 
```suggestion
                   
```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 6]
```diff
@@ -54,7 +56,7 @@ const DnsenumTool = () => {
         "Step 7: Select Reverse Lookup if required.\n";
     const sourceLink = "https://www.kali.org/tools/dnsenum/";
     const tutorial = "https://www.kali.org/tools/dnsenum/";
-    const dependencies = ["dnsenum"]; // Dependencies required by the component
+    const dependencies = ["dnsenum", "kill"]; // Dependencies required by the component
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:59]: Should not require kill
```suggestion
    const dependencies = ["dnsenum"]; // Dependencies required by the component
```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 7]
```diff
@@ -32,6 +32,8 @@ const DnsenumTool = () => {
     const [pid, setPid] = useState("");
     const [allowSave, setAllowSave] = useState(false);
     const [hasSaved, setHasSaved] = useState(false);
+    const [missingCommands, setMissingCommands] = useState<string[]>([]);
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:35]: The checkAllCommandsAvailability and installationModal components require some specific variables. 
```suggestion
    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
    
    
```

File: src/components/DNSenumTool/DNSenumTool.tsx [hunk 8]
```diff
@@ -32,6 +32,8 @@ const DnsenumTool = () => {
...
+    const [modalOpened, setModalOpened] = useState(false);
```
Comment (michaeljpigott) [inline src/components/DNSenumTool/DNSenumTool.tsx:36]: This will not be required once you use the pre-made component. 
```suggestion
  
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @junedtk632 

Nice attempt but I think you may have made this a bit more difficult than it needs to be. DDT has already got components for checking command availability and the installation model, so I recommend you use those. 

Kind regards

Michael

**Review by michaeljpigott — APPROVED** [review]
Hi @junedtk632 

Thank you for making those changes. The tool looks good and I approve the pull request. 

Kind regards

Michael

**Pull request merged**